src/components/TableStateContainer.tsx

```diff
@@ -5,31 +5,35 @@ import {ColumnDefinition} from './Table';
 import Pagination from './Pagination';
 import Checkbox from './Checkbox';
 import TableSearchBox from './TableSearchBox';
-import { connect, useSelector } from "react-redux";
-import { getSelectedOrganisation } from '../reducers'
-import { withRouter } from "react-router-dom";
-import {  injectIntl } from "react-intl";
+import { useSelector } from "react-redux";
+import { getSelectedOrganisation } from '../reducers';
 import {DataRetrievalState} from '../types/retrievingDataTypes';
 import unorderedIcon from "../images/list_order_icon_unordered.svg";
 import orderedIcon from "../images/list_order_icon_ordered.svg";
 import styles from './Table.module.css';
 import buttonStyles from '../styles/Buttons.module.css';
 
+interface checkboxAction {
+  displayValue: string,
+  actionFunction: Function,
+  nameOfDependentDataField?: string,
```

Comment:
I am thinking of another name for this variable, for example: `queryDataField`. What do you think?

Reply:
Hi Hoan,
I think this is done almost right.
I think a more scalable improvement might be that "nameOfDependentDataField" would instead be a function that takes the whole table row and returns a boolean. It could be called something like "testIfApplicableFunction".
For the rest i think this looks all very good!
kr Tom

Reply:
Thanks for your comment! Sounds like a very scalable improvement. I added this function to the `checkboxActions` of the ScenarioTable:

`checkIfActionIsApplicable: (row: any) => row.has_raw_results === true`